cfme/utils/appliance/__init__.py

```diff
@@ -1851,6 +1851,81 @@ def fqdn(self):
 
+    def get_disabled_regions(self, provider=None):
+        """Fetch appliance advanced config, get disabled regions for given provider's type
+
+        Only relevant for cloud providers azure and ec2 at the moment
+
+        Args:
+            provider: A BaseProvider object with settings_key attribute
+
+        Returns:
+            Default: Dict of ems_<provider> keys and values of disabled_regions map
+            when provider given: disabled_regions list from config
+            when no matching config found: None
+        """
+        ems_config = self.get_yaml_config()['ems']
+        if provider:
+            try:
+                prov_config = ems_config.get(getattr(provider, 'settings_key', None), {})  # safe
+                regions = prov_config['disabled_regions']  # KeyError
+            except KeyError:
+                logger.warning('Provider settings_key "%s" or "disabled_regions" '
+                               'not found in config %s', provider.settings_key, ems_config)
+                return None
+        else:
+            regions = {ems_key: yaml['disabled_regions']
+                       for ems_key, yaml in ems_config.items()
+                       if 'disabled_regions' in yaml}
+
+        return regions
+
+    def _modify_region_config(self, settings_key, value):
+        """Method for use by disable_regions and enable_all_regions to make yaml changes"""
+        yaml_conf = self.get_yaml_config()
+        try:
+            yaml_conf['ems'][settings_key]['disabled_regions'] = value
+        except KeyError:
+            # catches bad settings_keys and 'None'
+            logger.exception('Provider settings_key "%s" not found in config %s',
+                           settings_key, yaml_conf['ems'])
+            return False
+
+        self.set_yaml_config(yaml_conf)
+        return True
+
+    def set_disabled_regions(self, provider, *regions):
+        """Modify config to set disabled regions  to given items for the given provider's type
+
+        Only relevant for cloud providers azure and ec2 at the moment
+
+        Does NOT append to the list of disabled regions, SETS it. build your own darn list
+
+        Args:
+            provider: A BaseProvider object with settings_key attribute
+            *regions: none, one or many region names, on None enables all regions for provider type
+
+        Returns:
+            Boolean, result of update
+        """
+        assert self._modify_region_config(provider.settings_key, list(regions))
+        return self.get_disabled_regions(provider) == list(regions)
+
+    def enable_all_regions(self, provider):
+        """Modify appliance advanced settings to enable all regions for given provider's type
+
+        Only relevant for cloud providers azure and ec2 at the moment
+
+        Args:
+            provider: A BaseProvider object with settings_key attribute
+
+        Returns:
+            Boolean, True if settings changed to enable regions
```

Comment:
**required** the docstring is not entirely correct - if the settings kex is not found there will be a exception

Reply:
The exception is caught and `False` returned though - at least for `KeyError`, so I think this docblock is accurate, unless some other exception occurs.